crates/forge_fs/src/read_range.rs

```diff
@@ -0,0 +1,209 @@
+use std::cmp;
+use std::path::Path;
+
+use anyhow::Result;
+
+use crate::error::ForgeFileError;
+use crate::file_info::FileInfo;
+
+impl crate::ForgeFS {
+    /// Reads a specific range of characters from a file.
+    ///
+    /// Returns a tuple containing:
+    /// - The file content as a UTF-8 string.
+    /// - FileInfo containing metadata about the read operation including
+    ///   character positions.
+    pub async fn read_range_utf8<T: AsRef<Path>>(
+        path: T,
+        start_char: Option<u64>,
+        end_char: Option<u64>,
+    ) -> Result<(String, FileInfo)> {
+        let path_ref = path.as_ref();
```

Comment:
We should create a file handle in this function, and pass than handle to all the utility functions so that we don't keep opening the file again and again.